cmd/vspd/main.go

```diff
@@ -28,122 +30,143 @@ const maxVoteChangeRecords = 10
 const consistencyInterval = 30 * time.Minute
 
 func main() {
+	// Load config file and parse CLI args.
+	cfg, err := loadConfig()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "loadConfig error: %v\n", err)
+		os.Exit(1)
+	}
+
+	vspd, err := newVspd(cfg)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "newVspd error: %v\n", err)
+		os.Exit(1)
+	}
+
 	// Run until an exit code is returned.
-	os.Exit(run())
+	os.Exit(vspd.run())
 }
 
-// run is the main startup and teardown logic performed by the main package. It
-// is responsible for parsing the config, creating dcrd and dcrwallet RPC clients,
-// opening the database, starting the webserver, and stopping all started
-// services when a shutdown is requested.
-func run() int {
+type vspd struct {
+	cfg         *config
+	log         slog.Logger
+	shutdownCtx context.Context
```

Comment:
Contexts shouldn't be stored in structs.   

See the [stdlib context doco](https://pkg.go.dev/context):

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:

Reply:
Fixed